src/RevSysCalls.cc

```diff
@@ -1217,8 +1217,8 @@ EcallStatus RevCore::ECALL_clock_gettime() {
   }
   memset( &src, 0, sizeof( *tp ) );
   SimTime_t x = timeConverter->convertToCoreTime( Stats.totalCycles );
-  src.tv_sec  = x / 1000000000000ull;
-  src.tv_nsec = ( x / 1000 ) % 1000000000ull;
+  src.tv_sec  = time_t( x / 1000000000000 );
```

Comment:
Suffixes are not needed for constants except in specific situations. A constant ([integer literal](https://en.cppreference.com/w/cpp/language/integer_literal)) will be of the smallest type capable of holding its value. See the linked page for the priority of types based on the base and suffix.

Also, we don't know that `uint64_t` is necessarily the same type as `ull`. Best to avoid suffixes, and then the type of the constant will be the smallest type which can hold its value, and if it's stored in a variable, the conversions are well-defined.

When constants of a very specific type are needed, I use uniform initialization syntax such as `uint64_t{1}`. The braces `{}` ensure that the value is not narrowed (truncated) by the type conversion, while parentheses `uint64_t(1)` would allow narrowing and are a function-style cast with the same meaning as `((uint64_t) 1)`.